.github/workflows/codeql-analysis.yml

```
@@ -14,6 +14,9 @@ on:
     #        *  * * * *
     - cron: '30 1 * * *'
 
+permissions:
+  contents: read
+
 jobs:
   CodeQL-Build:
     runs-on: ubuntu-latest
```

thx @pichlermarc! just moving the permissions to be job scoped, which makes OSSF scorecard happier
```suggestion
  CodeQL-Build:
    permissions:
      security-events: write # for github/codeql-action/analyze to upload SARIF results
    runs-on: ubuntu-latest
```